common/compositor/renderstate.cpp

```diff
@@ -68,16 +68,8 @@ void RenderState::ConstructState(std::vector<OverlayLayer> &layers,
       }
       case HWCTransform::kTransform90: {
```

Comment:
I think right way would to move this kTransform90 case and handle it separately here. if (transform & kTransform90) {
/// blah blah
} else {
existing switch statement without kTransform90 case ?

Reply:
Maybe, but that would go a step beyond dead code removal. We've never handled the case where both kTransform90 and kReflectX have both been set, and so I wouldn't expect there to be any meaningful test coverage. I would suggest we clean this up now, and then add a separate issue for supporting that functionality.

Reply:
You can check here: https://github.com/intel/IA-Hardware-Composer/blob/master/os/android/iahwc2.cpp#L965 And now if I remember in some rotation modes, we had some layers rendered in-correctly which would be explained by this. My suggestion is fix already now as we have the plumbing in rest of the stack.

Reply:
> we had some layers rendered in-correctly which would be explained by this

Was that buggy behavior ever reported?

> My suggestion is fix already now as we have the plumbing in rest of the stack.

I'm not saying we shouldn't try to fix the rotation issues now, only that the scope of this PR and the issue that it fixes is the removal of unreachable code. No more. No less.

I'm happy to investigate, identify any relevant test cases, and then open a separate issue for the rotation mode behavior before we close #277.

Reply:
hmm No, removing the code is not really the scope of #277. If that was the fix than yes, but it's clear that it's not the fix(i.e. If we didnt take those into account anywhere else, ya what u did here made sense). Yes bugs where reported in our internal database.

Reply:
Dont confuse with what coverty is reporting, scope of #277 was to find the right fix, not just follow what the tool recommended.

Reply:
I don't think coverity recommended any particular course of action, it just states that this bit of code is dead, and that is all that was included in the text of #277. The right fix for dead code is to remove it, as it only serves to confuse people who are trying to read it. I would even go as far as saying that dead code is worse for readability than outdated comments.

The discovery (or rediscovery) of an underlying bug is a fortunate but secondary side effect of the process of going through a code cleanup. That is really the essence of why I'm pushing back on this, I want it to be acceptable for people to submit code cleanups.

I'm not trying to diminish our responsibility to report bugs and in this case due diligence requires that we open an issue, detailing the transform cases that obviously aren't being covered.

As I already stated, I'm fine with keeping #277 open until we fix all the issues that fall out of this exercise, but I really hope we don't set a precedent that code cleanups are unwelcome.

Reply:
I dont think we are setting any precedence for code clean ups, Code clean ups doesn't mean removing code :)  This is a bug for me where we didn't notice that these are bit flags. (Copy paste typos) If someone is doing code clean up, I would expect him to fix this as part of code clean up. This is by no means dead code....  

I would expect to file a feature in case transforms are not supported, they are today.

Reply:
So in other words what I am saying is, yes we did have buggy code and now let's not continue doing it but fix it with have proper logic to handle these cases. As far as your test cases, thats a genuine concern but we don't have tests for these kind of features today and that's something we should start building.

Reply:
> If someone is doing code clean up, I would expect him to fix this as part of code clean up.

Then that is precisely what we disagree on, and I won't push the topic any further. Just a difference in expectations.

> This is by no means dead code....

All I mean by dead code is that it is unreachable, and can be safely removed without any affect to the final program. Whatever context contributes to it being there in the first place doesn't make it any less dead.

Reply:
Fair enough :)